flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/flip1/RestartBackoffTimeStrategyFactoryLoader.java

```diff
@@ -0,0 +1,398 @@
+		if (restartStrategyName == null) {
+			return tryAdaptingClusterConfigurationOfDeprecatedValues(clusterConfiguration, adaptedConfiguration);
+		}
+
+		switch (restartStrategyName.toLowerCase()) {
+			case "none":
+			case "off":
+			case "disable":
+				setNoRestartStrategyToConfiguration(adaptedConfiguration);
+				return true;
+			case "fixeddelay":
+			case "fixed-delay":
+				adaptClusterConfigurationOfFixedDelayRestartStrategy(
+					clusterConfiguration,
+					adaptedConfiguration);
+				return true;
+			case "failurerate":
+			case "failure-rate":
+				adaptClusterConfigurationOfFailureRateRestartStrategy(
+					clusterConfiguration,
+					adaptedConfiguration);
+				return true;
+			default:
+				return false;
+		}
+	}
+
+	private static boolean tryAdaptingClusterConfigurationOfDeprecatedValues(
+		final Configuration clusterConfiguration,
+		final Configuration adaptedConfiguration) throws Exception {
+
+		// support deprecated ConfigConstants values
+		final int maxAttempts = clusterConfiguration.getInteger(
+			ConfigConstants.RESTART_STRATEGY_FIXED_DELAY_ATTEMPTS,
+			ConfigConstants.DEFAULT_EXECUTION_RETRIES);
+		final String pauseString = clusterConfiguration.getString(
+			AkkaOptions.WATCH_HEARTBEAT_PAUSE);
+		final String delayString = clusterConfiguration.getString(
+			ConfigConstants.RESTART_STRATEGY_FIXED_DELAY_DELAY,
+			pauseString);
+
+		final long delay;
+		try {
+			delay = Duration.apply(delayString).toMillis();
+		} catch (NumberFormatException nfe) {
+			if (delayString.equals(pauseString)) {
+				throw new Exception("Invalid config value for " +
+					AkkaOptions.WATCH_HEARTBEAT_PAUSE.key() + ": " + pauseString +
+					". Value must be a valid duration (such as '10 s' or '1 min')");
+			} else {
+				throw new Exception("Invalid config value for " +
+					ConfigConstants.RESTART_STRATEGY_FIXED_DELAY_DELAY + ": " + delayString +
+					". Value must be a valid duration (such as '100 milli' or '10 s')");
+			}
+		}
+
+		if (maxAttempts > 0 && delay >= 0) {
+			setFixedDelayRestartStrategyToConfiguration(adaptedConfiguration, delay, maxAttempts);
+			return true;
+		} else {
+			return false;
+		}
+	}
+
+	private static void adaptClusterConfigurationOfFixedDelayRestartStrategy(
+		final Configuration clusterConfiguration,
+		final Configuration adaptedConfiguration) throws Exception {
+
+		final int maxAttempts = clusterConfiguration.getInteger(
+			ConfigConstants.RESTART_STRATEGY_FIXED_DELAY_ATTEMPTS,
+			1);
+		final String delayString = clusterConfiguration.getString(
+			ConfigConstants.RESTART_STRATEGY_FIXED_DELAY_DELAY);
+
+		final long delay;
+		try {
+			delay = Duration.apply(delayString).toMillis();
+		} catch (NumberFormatException nfe) {
+			throw new Exception("Invalid config value for " +
+				ConfigConstants.RESTART_STRATEGY_FIXED_DELAY_DELAY + ": " + delayString +
+				". Value must be a valid duration (such as '100 milli' or '10 s')");
+		}
```

Comment:
code in line 284~ 291 is a duplicate to that in line 312~ 327, which could be extracted into a static function.

Reply:
Yes. we can extract the duration parsing logic.